tests/src/Context/DrupalContext.php

```diff
@@ -209,4 +216,111 @@ public function theFollowingContentsUsingForWysiwygFields($text_format, TableNod
     return $nodes;
   }
 
+  /**
+   * Add a field to a specified content type.
+   *
+   * @param string $arg1
+   *    The machine name of the content type to which attaching the field.
+   * @param TableNode $settings
+   *    Two columns table containing field settings.
+   *
+   * @Given a field with the following settings is added to :arg1 type:
+   *
+   * @throws \InvalidArgumentException
+   *    If the field exist already in the Drupal system.
+   */
+  public function aFieldWithTheFollowingSettingsIsAddedToType($arg1, TableNode $settings) {
+    $field_info = array();
+    $field_instance_info = array();
+
+    $field_instance_info['entity_type'] = 'node';
+    $field_instance_info['bundle'] = $arg1;
+
+    // Assign fields to user before creation.
+    foreach ($settings->getRowsHash() as $setting_name => $value) {
+      switch ($setting_name) {
+        case 'Label':
+          $field_instance_info['label'] = $value;
+          break;
+
+        case 'Name':
+          $field_name = 'field_' . $value;
+          $this->createFieldList[$field_name] = array(
```

Comment:
Maybe, we could add this field in the array only after the foreach. It's more simple to understand the logic and we limit errors by bad settings.